targets/ingester/src/cli.js

```diff
@@ -0,0 +1,155 @@
+    const pkgData = (await readFile(pkgInfoPath)).toString();
+    const pkgInfo = JSON.parse(pkgData);
+    return semver.lt(pkgInfo.version, latest);
+  } catch (error) {
+    console.error("[isPkgOutdated]", error);
+    return true;
+  }
+}
+/**
+ *
+ * @param {string} pkgName
+ * @param {string} url
+ */
+async function download(pkgName, url) {
+  return new Promise((resolve, reject) => {
+    getUri(url, function (err, rs) {
+      // !rs is only here so typescript won't complain
+      if (err || !rs) {
+        reject(`Error while downloading package ${pkgName} - ${url}`);
+        return;
+      }
+      rs.pipe(gunzip())
+        .pipe(
+          tar.extract(getPkgPath(pkgName), {
+            map: function mapHeader(header) {
+              // npm tarball have a root directory called /package so we remove it when extracting
+              header.name = header.name.replace("package/", "");
+              return header;
+            },
+          })
+        )
+        .on("finish", resolve);
+    });
+  });
+}
+
+/** @type {[string, (pkgName:string)=>Promise<import("./index.js").CdtnDocument[]>][]} */
+const dataPackages = [
+  ["@socialgouv/contributions-data", getContributionsDocuments],
+  ["@socialgouv/kali-data", getAgreementDocuments],
+  ["@socialgouv/legi-data", getCdtDocuments],
+  ["@socialgouv/fiches-vdd", getFichesServicePublic],
+  ["@socialgouv/fiches-travail-data", getFicheTravailEmploi],
+];
+/**
+ *
+ * @param {string} pkgName
+ * @param {string} pkgVersion
+ */
+async function getPackage(pkgName, pkgVersion = "latest") {
+  const pkgInfo = await got(
+    `http://registry.npmjs.org/${pkgName}/${pkgVersion}`
+  ).json();
+  const url = pkgInfo.dist.tarball;
+  const latest = pkgInfo.version;
+  if (await isPkgOutdated(pkgName, latest)) {
+    await download(pkgName, url);
+  }
+}
+
+async function main() {
+  for (const [pkgName] of dataPackages) {
+    await getPackage(pkgName);
+  }
+  if (args.dryRun) {
+    console.log("dry-run mode");
+  }
+  for (const [pkgName, getDocument] of dataPackages) {
+    if (!getDocument) {
+      continue;
+    }
+    const documents = await getDocument(pkgName);
+    if (args.dryRun) {
+      continue;
+    }
+    console.log(`ready to ingest ${pkgName}`);
+    for (const doc of documents) {
+      const { id, title, text, slug, source, ...document } = doc;
+      const result = await client
+        .mutation(insertDocumentsMutation, {
+          document: {
+            cdtn_id: generateCdtnId(`${source}${id}`),
+            document,
+            initial_id: id,
+            meta_description: document.description || "",
+            slug,
+            source,
+            text,
+            title,
+          },
+        })
+        .toPromise();
+
+      if (result.error) {
+        console.error(result.error);
+        throw new Error(`insert document alert ${title}`);
+      }
+    }
+  }
+}
+
+main()
+  .then(() => console.log(`Finish Ingest`))
+  .catch(console.error);
```

Comment:
we could extract CLI logic from dowload/update code

Reply:
Could be useful to have the possibility to decide which data package we want to update ?

Reply:
Since it as initially design this way I could be simple to make the change. My concerns was how to automate it after in k8s land.